axelor-human-resource/src/main/java/com/axelor/apps/hr/web/leave/LeaveController.java

```diff
@@ -438,4 +448,48 @@ public String leaveValidateMenuTag() {
     return Beans.get(HRMenuTagService.class)
         .countRecordsTag(LeaveRequest.class, LeaveRequestRepository.STATUS_AWAITING_VALIDATION);
   }
+
+  @SuppressWarnings("unchecked")
+  public void exportLeaveRequest(ActionRequest request, ActionResponse response) {
+    try {
+
+      Context context = request.getContext();
+      List<Long> ids = null;
+
+      if (context.get("_ids") == null) {
+        JpaSecurity security = Beans.get(JpaSecurity.class);
+        ids =
+            Criteria.parse(request)
+                .createQuery(
+                    LeaveRequest.class,
+                    security.getFilter(JpaSecurity.CAN_READ, LeaveRequest.class))
+                .fetch().stream()
```

Comment:
Please, instead of using a stream to get the id, modify the query to `SELECT` and retrieve directly only the ids, to improve performances.